### PR TITLE
Add Control.Pipeline (|>, <|)

### PR DESCRIPTION
**Pull request by lpil**
Closes https://github.com/idris-lang/Idris-dev/issues/2817

Unfinished- I was unsure where to put the tests for this. Could you point me in the right direction? :)

Thanks,
Louis

### CODE REVIEW

**Comment by ahmadsalim** [comment]
I do not think it's a large enough feature, that you need to make tests for it. It will type-check code snippets in docs anyway, so just add some example for the `<|` as well :).

File: libs/contrib/Control/Pipeline.idr [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+infixr 0 <|
+
+%access public export
+
+
+||| Pipeline style function application, useful for chaining
+||| functions into a series of transformations, reading top
+||| to bottom.
+|||
+|||     [[1], [2], [3]] |> join |> map (* 2)
```
Comment (ahmadsalim) [inline libs/contrib/Control/Pipeline.idr]: you should surround this with idris example blocks as follows:

    ```idris example
     [[1], [2], [3]] |> join |> map (* 2)
    ```

Comment (lpil) [inline libs/contrib/Control/Pipeline.idr]: Thanks :)

**Comment by lpil** [comment]
No tests? 😱  This really is unfamiliar territory for me :P

I'll fix up the docs.

File: libs/contrib/Control/Pipeline.idr [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module Control.Pipline
```
Comment (ozgurakgun) [inline libs/contrib/Control/Pipeline.idr]: There is a typo in the module name.

Comment (lpil) [inline libs/contrib/Control/Pipeline.idr]: Thank you.

**Comment by lpil** [comment]
Typo fixed and examples added.

**Comment by ahmadsalim** [comment]
@lpil I mean, we do have a type system after all :). I think adding tests is generally useful, but perhaps not necessary for such simple features.

**Comment by david-christiansen** [comment]
> It will type-check code snippets in docs anyway

Note that a failure to type check in docs does not make the module fail to type check. This is so docs can exhibit examples of errors that types rule out.

**Comment by ahmadsalim** [comment]
Thanks for the PR.

**Comment by lpil** [comment]
Thanks!

**Pull request merged**